engine/include/cubos/engine/ui/text/text.hpp

```diff
@@ -0,0 +1,48 @@
+/// @file
+/// @brief Component @ref cubos::engine::UIText.
+/// @ingroup ui-text-plugin
+
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include <glm/vec4.hpp>
+
+#include <cubos/core/gl/render_device.hpp>
+#include <cubos/core/reflection/reflect.hpp>
+
+#include <cubos/engine/api.hpp>
+#include <cubos/engine/assets/asset.hpp>
+#include <cubos/engine/assets/assets.hpp>
+#include <cubos/engine/font/atlas.hpp>
+#include <cubos/engine/font/font.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Component used to draw text on the UI.
+    ///
+    /// @ingroup ui-text-plugin
+    struct CUBOS_ENGINE_API UIText
+    {
+        CUBOS_REFLECT;
+
+        /// @brief The text of the element.
+        std::string text;
+
+        /// @brief The color of the text.
+        glm::vec4 color{1};
+
+        /// @brief The font size to draw the characters.
+        float fontSize{24.0F};
+
+        /// @brief The font to be used.
+        Asset<Font> font{AnyAsset("93cbe82e-9c9b-4c25-aa55-5105c1afd0cc")};
+
+        core::gl::VertexArray va{nullptr};
+
+        std::shared_ptr<FontAtlas> atlas{nullptr};
+
+        size_t vertexCount{0};
```

Comment:
If the user isn't meant to mess with those these should be private (and the struct made a class) I think :thinking: otherwise it gets kinda confusing. You can add methods so that you can interact with them from the plugin itself